src/NServiceBus.Core.Tests/Pipeline/Incoming/InvokeHandlerTerminatorTest.cs

```diff
@@ -126,11 +148,20 @@ static ActiveSagaInstance AssociateSagaWithMessage(FakeSaga saga, IInvokeHandler
 
         MessageHandler CreateMessageHandler(Action<object, object, IMessageHandlerContext> invocationAction, object handlerInstance)
         {
+            return CreateMessageHandler(invocationAction, handlerInstance, string.Empty, () => TaskEx.CompletedTask);
+        }
+
+        MessageHandler CreateMessageHandler(Action<object, object, IMessageHandlerContext> invocationAction, object handlerInstance, string methodName, Func<Task> handlerTask)
+        {
             var messageHandler = new MessageHandler((instance, message, handlerContext) =>
             {
                 invocationAction(instance, message, handlerContext);
-                return TaskEx.CompletedTask;
-            }, handlerInstance.GetType())
+                return handlerTask();
+            }, handlerInstance.GetType(), new FakeMethodInfo(handlerInstance, methodName))
+            //{
```

Comment:
Allows for friendly exception messages, see below. The built in MethodInfo classes weren't a good fit, so I just implemented it for the test.